AccountsService.get_by_address() couple of fixes

AndreMiras opened the pull request:
Adds AccountsService.get_by_address() fixes and unit tests:

- fixes TypeError
- added two unit tests

AndreMiras commented:
Could we merge this before it gets outdated and full of conflicts?

janx commented on pyethapp/accounts.py:
```diff
@@ -480,9 +480,7 @@ def get_by_address(self, address):
         if len(accounts) == 0:
-            raise KeyError('account not found by address', address=address.encode('hex'))
-        elif len(accounts) > 1:
-            log.warning('multiple accounts with same address found', address=address.encode('hex'))
```
Why is this multiple accounts case removed?

AndreMiras replied:
My bad, I thought it was dead code because I thought it was prevented upfront here:
https://github.com/ethereum/pyethapp/blob/v1.5.0/pyethapp/accounts.py#L283
Unit test (test_add_account_twice):
https://github.com/ethereum/pyethapp/blob/v1.5.0/pyethapp/tests/test_account_service.py#L104
But that's only on known uuid, so let me update the PR.

AndreMiras replied:
OK I've reverted it. I wanted to covert it with a test, but it requires us to introduce pytest mocks.
So it will be for next time.

AndreMiras commented:
I've added the coverage test for multiple accounts case.
It requires `pytest-mock` additional requirement.

AndreMiras commented:
Hi @janx,
Anything I could do to help speeding up the merging? Or anything you want me to change from this pull request?

janx commented on requirements.txt:
```diff
@@ -18,3 +18,5 @@ pbkdf2
 pexpect
 pyelliptic==1.5.7
+mock==2.0.0
+pytest-mock==1.6.0
```
It would be better to keep this dev dependencies in a seperate file, like what pyethereum does.

janx commented:
@AndreMiras can you also squash these commits into a single one?

AndreMiras commented:
Hi @janx thank you for your feedback.
I've updated my pull request with:

 * dedicated `dev_requirements.txt` file
 * squashed into single commit

AndreMiras committed:
AccountsService.get_by_address() couple of fixes

Fix TypeError on get_by_address() with no match
The error was:
TypeError: exceptions.KeyError does not take keyword arguments

Also unit test get_by_address() multiple matches and no match.
Introduced `dev_requirements.txt` file.
Increases coverage and requires pytest-mock additional dep.

janx commented:
@AndreMiras thanks :beer:
